examples/http/src/main/java/io/opentelemetry/example/HttpClient.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.opentelemetry.example;
+
+import io.opentelemetry.context.propagation.HttpTextFormat;
+import io.opentelemetry.exporters.inmemory.InMemorySpanExporter;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.trace.SpanData;
+import io.opentelemetry.sdk.trace.TracerSdkFactory;
+import io.opentelemetry.sdk.trace.export.SimpleSpansProcessor;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.Status;
+import io.opentelemetry.trace.Tracer;
+import io.opentelemetry.trace.propagation.HttpTraceContext;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+public class HttpClient {
+
+  static int port = 8080;
+  // OTel API
+  Tracer tracer;
+  // Export traces in memory
+  InMemorySpanExporter inMemexporter = InMemorySpanExporter.create();
+  // Share context via http headers
+  HttpTextFormat<SpanContext> textFormat = new HttpTraceContext();
+  // Inject the span context into the request
+  HttpTextFormat.Setter<HttpURLConnection> setter =
+      new HttpTextFormat.Setter<HttpURLConnection>() {
+        @Override
+        public void put(HttpURLConnection carrier, String key, String value) {
+          carrier.setRequestProperty(key, value);
+        }
+      };
+
+  private void initTracer() {
+    // Get the tracer
+    TracerSdkFactory tracerFactory = OpenTelemetrySdk.getTracerFactory();
+    // Set to process in memory the spans
+    tracerFactory.addSpanProcessor(SimpleSpansProcessor.newBuilder(inMemexporter).build());
+    // Give a name to the traces
+    this.tracer = tracerFactory.get("io.opentelemetry.example.HttpClient");
+  }
+
+  public HttpClient() throws Exception {
+    initTracer();
+
+    // Connect to the server locally
+    URL url = new URL("http://127.0.0.1:" + port);
+    HttpURLConnection con = (HttpURLConnection) url.openConnection();
+
+    // Start a span
+    Span span = tracer.spanBuilder("Request hello API").startSpan();
+    // Inject the request with the context
+    textFormat.inject(span.getContext(), con, setter);
+
+    // Process the request
+    con.setRequestMethod("GET");
+    int status = con.getResponseCode();
```

Comment:
This should be embedded in a `try` block and a `catch` block should be used for reporting errors by setting the span status accordingly.